HeavyFlavorAnalysis/Onia2MuMu/src/Onia2MuMuPAT.cc

```diff
@@ -136,10 +138,10 @@ Onia2MuMuPAT::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
 	CachingVertex<5> VtxForInvMass = vtxFitter.vertex( t_tks );
 
         Measurement1D MassWErr(jpsi.M(),-9999.);
-        try {
-	  MassWErr = massCalculator.invariantMass( VtxForInvMass, muMasses );
-        } catch (std::exception & err) {
-          std::cout << " Field is 0, invalidating vertex " << std::endl;
+        if ( field->nominalValue() > 0 ) {
+          MassWErr = massCalculator.invariantMass( VtxForInvMass, muMasses );
+        } else {
+          std::cout << " Field is " << field->nominalValue() << " kGauss, invalidating vertex combination" << std::endl;
```

Comment:
sorry i missed the cout  which should be removed- I guess you don't want to print anything this module should work routinely with B=0 (even if we hope not to have any such data!)

@alberto-sanchez